pkg/da/data_availability_header.go

```diff
@@ -167,6 +167,14 @@ func (dah *DataAvailabilityHeader) IsZero() bool {
 	return len(dah.ColumnRoots) == 0 || len(dah.RowRoots) == 0
 }
 
+// SquareSize returns the number of rows in the original data square.
+func (dah *DataAvailabilityHeader) SquareSize() (int, error) {
+	if len(dah.RowRoots)%2 != 0 {
```

Comment:
[optional] If we aim to do some sanity checks like this, then we may as well check whether the number of `RowRoots` and `ColRoots` are the same.

Reply:
Removed sanity checks b/c @Wondertan proposed removing the error return parameter